schemas/context/profile-contact-crm.schema.json

```diff
@@ -0,0 +1,154 @@
+          "description": "Field recording what source the lead originated from.",
+          "type": "string"
+        },
+        "xdm:contactDescription": {
+          "title": "Contact Description",
+          "description": "Contact description.",
+          "type": "string"
+        },
+        "xdm:contactStatus": {
+          "title": "Contact Status",
+          "description": "Current status of the contact.",
+          "type": "string"
+        },
+        "xdm:contactStatusDetail": {
+          "title": "Contact Status Detail",
+          "description": "Additional Details of the current status of the contact.",
+          "type": "string"
+        },
+        "xdm:contactType": {
+          "title": "Contact Type",
+          "description": "Type of the contact.",
+          "type": "string"
+        },
+        "xdm:userID": {
+          "title": "User Id",
+          "description": "User id.",
+          "type": "string"
+        },
+        "xdm:originalReferrer": {
+          "title": "Original Referrer",
+          "description": "The original referrer is typically the page that leads to a form fill out.",
+          "type": "string"
+        },
+        "xdm:originalSearchEngine": {
+          "title": "Original Search Engine",
+          "description": "Search engine that the original referrer used.",
+          "type": "string"
+        },
+        "xdm:originalSearchPhrase": {
+          "title": "Original Search Phrase",
+          "description": "Search phrase used by the original referrer.",
+          "type": "string"
+        },
+        "xdm:originalSourceInfo": {
+          "title": "Original Source Info",
+          "description": "Original source info.",
+          "type": "string"
+        },
+        "xdm:originalSourceType": {
+          "title": "Original Source Type",
+          "description": "Original ource type.",
+          "type": "string"
+        },
+        "xdm:timezone": {
+          "title": "Timezone",
+          "description": "Timezone of the contact's location.",
+          "type": "string"
+        },
+        "xdm:emailOptOutReason": {
+          "title": "Email Opt Out Reason",
+          "description": "Reason for the contact opting out of recieving any emails if available.",
+          "type": "string"
+        },
+        "xdm:phoneOptOutReason": {
+          "title": "Phone Opt Out Reason",
+          "description": "Reason for the contact opting out of recieving any phone calls if available.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/extensible#/definitions/@context"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/external-source-system-audit-details"
+    },
+	{
+      "$ref": "https://ns.adobe.com/xdm/context/identitymap"
+    },	
+    {
+      "$ref": "#/definitions/profile-contact-crm"
+    }
+  ],
+  "required": ["xdm:contactID", "xdm:identityMap"],
```

Comment:
xdm: AccountID is a MUST property, we don't need orphan contacts.

Reply:
As per industry standards, in a massive data lake and data warehouse, we should not impose hard referential constraints on foreign keys as certain data might flow before the foreign key is populated in the source. enforcing the primary key definitely makes sense but enforcing the foreign key might cause data lags. Please suggest if you still want me to add Account to the required ?

Reply:
Explained the reasoning